src/System.Drawing.Common/src/System/Drawing/Icon.Unix.cs

```diff
@@ -707,181 +711,181 @@ private void InitFromStreamWithSize(Stream stream, int width, int height)
                 throw new ArgumentNullException(nameof(stream));
 
             if (stream.Length == 0)
-                throw new System.ArgumentException("The argument 'stream' must be a picture that can be used as a Icon", nameof(stream));
-
-            //read the icon header
-            BinaryReader reader = new BinaryReader(stream);
-
-            //iconDir = new IconDir ();
-            iconDir.idReserved = reader.ReadUInt16();
-            if (iconDir.idReserved != 0) //must be 0
-                throw new System.ArgumentException("Invalid Argument", nameof(stream));
-
-            iconDir.idType = reader.ReadUInt16();
-            if (iconDir.idType != 1) //must be 1
-                throw new System.ArgumentException("Invalid Argument", nameof(stream));
+                throw new ArgumentException(SR.Format(SR.InvalidPictureType, "picture", nameof(Icon)));
 
-            ushort dirEntryCount = reader.ReadUInt16();
-            imageData = new ImageData[dirEntryCount];
-            iconDir.idCount = dirEntryCount;
-            iconDir.idEntries = new IconDirEntry[dirEntryCount];
             bool sizeObtained = false;
-            // now read in the IconDirEntry structures
-            for (int i = 0; i < dirEntryCount; i++)
+            ushort dirEntryCount;
+            // Read the icon header
+            using (BinaryReader reader = new BinaryReader(stream))
             {
-                IconDirEntry ide;
-                ide.width = reader.ReadByte();
-                ide.height = reader.ReadByte();
-                ide.colorCount = reader.ReadByte();
-                ide.reserved = reader.ReadByte();
-                ide.planes = reader.ReadUInt16();
-                ide.bitCount = reader.ReadUInt16();
-                ide.bytesInRes = reader.ReadUInt32();
-                ide.imageOffset = reader.ReadUInt32();
-
-                // Vista 256x256 icons points directly to a PNG bitmap
-                // 256x256 icons are decoded as 0x0 (width and height are encoded as BYTE)
-                // and we ignore them just like MS does (at least up to fx 2.0) 
-                // Added: storing data so it can be saved back
-                if ((ide.width == 0) && (ide.height == 0))
-                    ide.ignore = true;
-                else
-                    ide.ignore = false;
-
-                iconDir.idEntries[i] = ide;
-
-                //is this is the best fit??
-                if (!sizeObtained)
+                iconDir.idReserved = reader.ReadUInt16();
+                if (iconDir.idReserved != 0) //must be 0
+                    throw new ArgumentException(SR.Format(SR.InvalidPictureType, "picture", nameof(Icon)));
```

Comment:
Why `nameof(Icon)` and `nameof(stream)` below..

Reply:
Heh, the effect of copying and pasting from Icon.Windows.cs 😆